src/Particle/HDFWalkerInput_0_4.cpp

```diff
@@ -215,38 +220,53 @@ bool HDFWalkerInput_0_4::read_hdf5_scatter(const std::filesystem::path& h5name)
   using Buffer_t = std::vector<QMCTraits::RealType>;
 
   const int np1 = myComm->size() + 1;
-  std::vector<int> counts(myComm->size()), woffsets(np1, 0);
-  FairDivideLow(nw_in, myComm->size(), woffsets);
+  std::vector<int> woffsets_weights(np1, 0);
+  FairDivideLow(nw_in, myComm->size(), woffsets_weights);
 
-  const size_t nw_loc = woffsets[myComm->rank() + 1] - woffsets[myComm->rank()];
+  std::vector<int> counts_weights(myComm->size());
+  for (int i = 0; i < counts_weights.size(); ++i)
+    counts_weights[i] = woffsets_weights[i + 1] - woffsets_weights[i];
 
+  // walker counts and offsets for electron coordinates
+  std::vector<int> woffsets(woffsets_weights);
```

Comment:
I meant here
```
std::vector<int> woffsets(np1);
```
and below do
```
woffsets[i] = woffsets_weights[i] * nitems;
```
I feel this is more readable.

Reply:
done